stl/inc/xmemory

```diff
@@ -529,38 +583,47 @@ struct _Normal_allocator_traits { // defines traits for allocators
         }
     }
 
-    static void deallocate(_Alloc& _Al, pointer _Ptr, size_type _Count) {
+    static _CONSTEXPR20_DYNALLOC void deallocate(_Alloc& _Al, pointer _Ptr, size_type _Count) {
         _Al.deallocate(_Ptr, _Count);
     }
 
     template <class _Ty, class... _Types>
-    static void construct(_Alloc& _Al, _Ty* _Ptr, _Types&&... _Args) {
+    static _CONSTEXPR20_DYNALLOC void construct(_Alloc& _Al, _Ty* _Ptr, _Types&&... _Args) {
         if constexpr (_Uses_default_construct<_Alloc, _Ty*, _Types...>::value) {
             (void) _Al; // TRANSITION, DevCom-1004719
+#ifdef __cpp_lib_constexpr_dynamic_alloc
+            _STD construct_at(_Ptr, _STD forward<_Types>(_Args)...);
+#else // __cpp_lib_constexpr_dynamic_alloc
             ::new (static_cast<void*>(_Ptr)) _Ty(_STD forward<_Types>(_Args)...);
+#endif // __cpp_lib_constexpr_dynamic_alloc
         } else {
             _Al.construct(_Ptr, _STD forward<_Types>(_Args)...);
         }
     }
 
     template <class _Ty>
-    static void destroy(_Alloc& _Al, _Ty* _Ptr) {
+    static _CONSTEXPR20_DYNALLOC void destroy(_Alloc& _Al, _Ty* _Ptr) {
         if constexpr (_Uses_default_destroy<_Alloc, _Ty*>::value) {
+            (void) _Al;

```

Comment:
It appears that this occurrence in `destroy` is unnecessary, because there's an `if constexpr` branch using `_Al` - however, the occurrence in `deallocate` is necessary because the usage of `_Al` is guarded with `#ifdef __cpp_lib_constexpr_dynamic_alloc`. So that suppression should be preserved, without a `TRANSITION`. Checking all occurrences.

Reply:
This was the only unnecessary occurrence, the others were all necessary.